lib/controllers/widget/device_list.dart

```diff
@@ -0,0 +1,182 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:chrome_management_app/controllers/widget/device_summary.dart';
+import 'package:chrome_management_app/models/api_calls.dart';
+import 'package:chrome_management_app/objects/account_devices.dart';
+import 'package:chrome_management_app/objects/basic_device.dart';
+import 'package:flutter/material.dart';
+import 'package:http/http.dart' as http;
+
+import '../devices.dart';
+
+/// Stateful Widget to show a list of devices.
+class DeviceList extends StatefulWidget {
+  /// Authorization token to get information form the Directory API.
+  final String _authToken;
+
+  /// Constructor of Widget
+  DeviceList(this._authToken, {Key key}) : super(key: key);
+  @override
+
+  /// Create initial state of the Widget.
+  _DeviceListState createState() => _DeviceListState(_authToken);
+}
+
+/// DeviceList Widget states
+class _DeviceListState extends State<DeviceList> {
+  /// Authorization token to call the list of devices.
+  String _authToken;
+
+  /// Account information from Directory Chrome Devices API.
+  AccountDevices _list;
+
+  /// Verify that all devices from account are loaded.
+  bool _allDevicesLoaded = false;
+
+  /// Stores any error/exception description.
+  String _warning;
+
+  /// Verify if a error/exception happend while loading the list of devices.
+  bool _errorOnLoading = false;
+
+  /// Verify the list is loading.
+  bool _loading = false;
+
+  /// Constructor of state.
+  _DeviceListState(this._authToken);
+
+  /// Scroll controller to manage lazy load.
+  ScrollController _scrollController;
+
+  /// Initial state of the Widget.
+  @override
+  void initState() {
+    _scrollController = ScrollController();
+    super.initState();
+    _getInitalDevices();
+    // Add listener to scroll controller to load more devices when user scrolls
+    // 75% or more of the view height.
+    _scrollController.addListener(() {
+      double maxScroll = _scrollController.position.maxScrollExtent;
+      double currentScroll = _scrollController.position.pixels;
+      double delta = MediaQuery.of(context).size.height * 0.25;
```

Comment:
should make this a class constant

Reply:
It is a listener, this evaluations happen every time the user scrolls.